custom/icds/messaging/indicators.py

```diff
@@ -115,3 +121,61 @@ def get_messages(self, language_code=None):
         message = self.render_template(context, language_code=language_code)
 
         return [message]
+
+
+class LSSubmissionPerformanceIndicator(LSIndicator):
+    template = 'ls_no_submissions.txt'
+
+    def __init__(self, domain, user):
+        super(LSSubmissionPerformanceIndicator, self).__init__(domain, user)
+
+        child_locations = [l.location_id for l in self.user.sql_location.child_locations()]
+        self.aww_user_ids = user_ids_at_locations(child_locations)
+        self.last_submission_dates = get_last_submission_time_for_users(
+            self.domain, self.aww_user_ids, self.get_datespan()
+        )
+
+    def get_datespan(self):
+        today = datetime(self.now.year, self.now.month, self.now.day)
+        end_date = today + timedelta(days=1)
+        start_date = today - timedelta(days=30)
+        return DateSpan(start_date, end_date, timezone=self.timezone)
+
+    def should_send(self):
+        if len(self.aww_user_ids) != len(self.last_submission_dates):
+            # some users did not submit in the datespan
+            return True
+
+        now_date = self.now.date()
+        for user, date in self.last_submission_dates.items():
+            if not date or (now_date - date).days > 7:
+                return True
+
+        return False
+
+    def get_messages(self, language_code=None):
+        one_week_user_ids = []
+        one_month_user_ids = []
+
+        now_date = self.now.date()
+        for user_id in self.aww_user_ids:
+            last_sub_date = self.last_submission_dates.get(user_id)
+            if not last_sub_date:
+                one_month_user_ids.append(user_id)
+            else:
+                days_since_submission = (now_date - last_sub_date).days
+                if days_since_submission > 7:
+                    one_week_user_ids.append(user_id)
+
+        one_week_loc_ids = get_users_location_ids(self.domain, one_week_user_ids)
+        one_month_loc_ids = get_users_location_ids(self.domain, one_month_user_ids)
+        one_week_loc_names = SQLLocation.objects.filter(location_id__in=one_week_loc_ids).values_list('name', flat=True)
+        one_month_loc_names = SQLLocation.objects.filter(location_id__in=one_month_loc_ids).values_list('name', flat=True)
```

Comment:
This will take a different DB accessor that isn't written yet. I have something similar coming up in my next indicator so I will port this indicator to use that once I'm done with that.